src/vos/vos_aggregate.c

```diff
@@ -1364,11 +1374,23 @@ insert_segments(daos_handle_t ih, struct agg_merge_window *mw,
 		phy_ent = lgc_seg->ls_phy_ent;
 
 		if (phy_ent != NULL && !bio_addr_is_hole(&ent_in->ei_addr)) {
+			/*
+			 * the csum free flag indicates that memory has been allocated and should
+			 * only be allocated once for the phy_ent.
+			 */
+			if (phy_ent->pe_csum_free)
+				D_FREE(phy_ent->pe_csum_info.cs_csum);
```

Comment:
Minor: I don't think we need to free & re-alloc here, since the physical entry is always truncated (could be truncated multiple times when it spans several mw, though it's very unlikely), the mem allocation on first truncation should be large enough.